app/client/src/utils/index.ts

```diff
@@ -14,3 +14,12 @@ export function debounce<T extends Callback>(cb: T, timeout: number): Callback {
         }, timeout);
     };
 }
+export function isURL(str: string) {
+    const pattern = new RegExp('^(https?:\\/\\/)?'+
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ 
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ 
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ 
+      '(\\?[;&a-z\\d%_.~+=-]*)?'+ 
+      '(\\#[-a-z\\d_]*)?$','i'); 
+    return !!pattern.test(str);
```

Comment:
One more question on this, did you make this or find it somewhere?

Reply:
I found that pattern on one of the stackoverflow pages I believe; Is that something I should avoid?

Reply:
It's not something bad.  It's usually good to add a source link in the comments.  Also, maybe try testing out the regex on https://regexr.com/.  I think I found the SO post and different people say different things about URL detection.

Reply:
I just went ahead and pushed the updated regex. I looked deeper into regex and came across many different ones that people came up with. I found this particular one that passed all of the different possible combinations for a link I could think of. I also added the source link in the comments as you suggested!